sound/soc/intel/haswell/sst-haswell-dsp.c

```diff
@@ -250,14 +250,22 @@ static void hsw_set_dsp_D3(struct sst_dsp *sst)
 
 	/* Disable core clock gating (VDRTCTL2.DCLCGE = 0) */
 	reg = readl(sst->addr.pci_cfg + SST_VDRTCTL2);
-	reg &= ~(SST_VDRTCL2_DCLCGE | SST_VDRTCL2_DTCGE);
+	reg &= ~SST_VDRTCL2_DCLCGE;
 	writel(reg, sst->addr.pci_cfg + SST_VDRTCTL2);
 
-	/* enable power gating and switch off DRAM & IRAM blocks */
+	/* Stall and reset core */
+	sst_dsp_shim_update_bits_unlocked(sst, SST_CSR,
+					  SST_CSR_RST | SST_CSR_STALL,
+					  SST_CSR_RST | SST_CSR_STALL);
+
```

Comment:
@keyonjie This seems extra with regards to commit message. Probably good to mention why this is added.

Reply:
@kv2019i yes, let me add it, thanks.